python/cudf/cudf/core/dataframe.py

```diff
@@ -1319,13 +1320,12 @@ def _slice(self: T, arg: slice) -> T:
         ]
         result = self._from_columns_like_self(
             libcudf.copying.columns_slice(columns_to_slice, [start, stop])[0],
-            self._column_names,
+            [*self._column_names],
```

Comment:
This is a fix to make mypy work. Seems strange that mypy doesn't catch this in the previous PR?

Reply:
Or just change the annotation for `_column_names` to `Iterable[str]` instead of `List[str]`.

Reply:
Another thing I can do here is change the type annotation of "column_names" as iterables, since it's unclear whether the `column_names` are mutable or not.

Reply:
:man_facepalming: I wish github can refresh faster when I reply to this lol.